app/src/main/java/org/hackillinois/android/view/SplashScreenActivity.kt

```diff
@@ -0,0 +1,61 @@
+package org.hackillinois.android.view
+
+import android.content.Context
+import android.content.Intent
+import android.support.v7.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import org.hackillinois.android.App
+
+import org.hackillinois.android.R
+import org.hackillinois.android.model.User
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class SplashScreenActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        var jwt = loadJWT();
+
+        if(jwt != "") {
+            var api = App.getAPI(jwt)
+            api.user.enqueue(object: Callback<User> {
+                override fun onFailure(call: Call<User>, t: Throwable) {
+                    Log.e("LoginActivity", "Failed to check is jwt is valid")
+                }
+
+                override fun onResponse(call: Call<User>, response: Response<User>) {
+                    if(response.code() == 200) {
```

Comment:
If onResponse() gets called instead of onFailure() I believe it's assumed that the code is 200

Reply:
The description in the documentation seems to indicate that the response code could be non-200. https://square.github.io/retrofit/2.x/retrofit/retrofit2/Callback.html#onResponse-retrofit2.Call-retrofit2.Response-